docs/components/message-factory.md

```diff
@@ -0,0 +1,44 @@
+# Message Factory
+
+**Factory interfaces for PSR-7 HTTP Message.**
+
+
+## Rationale
+
+While it should be possible to use every PSR-7 aware HTTP client with any RequestInterface implementation,
+creating the request objects will still tie an application to a specific implementation.
+In the case of reusable libraries, an application could end up installing several request implementations.
```

Comment:
If each reusable library is tied to a specific message implementation, an application could en up installing several message implementations.